src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -75,9 +75,17 @@ private unsafe static int CompareOrdinalIgnoreCaseHelper(String strA, String str
 
         private unsafe static bool EqualsHelper(String strA, String strB)
         {
-            Contract.Requires(strA != null);
-            Contract.Requires(strB != null);
-            Contract.Requires(strA.Length == strB.Length);
+            Debug.Assert((strA == null || strB == null) || strA.Length == strB.Length);
+
+            if ((object)strA == strB)
```

Comment:
`List.Add` showed up on the radar in ASP.NET workload, and it was not the first time - I know people were considering doing something about it before.